text/0567-use-and-resources.md

```diff
@@ -0,0 +1,953 @@
+
+```ts
+declare function getUseInfo(): {
+  owner: Owner;
+  isCreating: boolean;
+  isUpdating: boolean;
+};
+```
+
+This information will be populated before calling the initializer, and can be
+called _exactly once_ during initialization. Calling a second time will throw an
+assertion, which is meant to prevent abuse of the API by other code (e.g. other
+classes that are initialized when creating the Resource). For safety, it should
+always be called immediately in `DEBUG` mode by any Resource implementation,
+even if it is not actually needed. Not calling it at all in `DEBUG` mode will
+also throw an assertion to help make sure implementers don't leak it
+accidentally to user code.
+
+#### Setting Arguments
+
+Additionally, initializers can only return one value, which is the manager
+handle itself. An important aspect of resources is that they should be able to
+respond to incoming arguments that are consumed while running the initializer,
+and in order to do this, the arguments have to be passed to the manager. In
+order to do this, users can use the `setUseArgs` method.
+
+```ts
+declare function setUseArgs(...args: unknown[]): void;
+```
+
+Like with `getUseInfo`, this function can be used _exactly once_ during the
+initializer's execution to save off the arguments, and not using it in DEBUG
+mode will throw an assertion to help make sure it isn't accidentally leaked to
+user code.
+
+#### Scheduling `@use` Initialization
+
+As mentioned above, the timing semantics of `@use` properties are that they
+start automatically, asynchronously, after the parent class they are defined on
+with is created, or when they are first consumed, whichever is first. Because of
+limitations of Ember's current stage 1/legacy decorators, this is not possible
+just by decorating a class element such as a field or method, since we intend to
+create a getter and not a field. Getters do not run any code on initialization.
+
+As such, Ember will provide a function that can be used to schedule starting any
+`@use` instances defined on the class, `scheduleUseStart()`
+
+```js
+class GlimmerComponent {
+  constructor() {
+    scheduleUseStart(this);
+  }
+}
+```
+
+This method only needs to be called once per instance of a class, and can be
+called by parent classes. It is generally intended to be an implementation
+detail. The following framework provided classes will use this function by
+default:
+
+- `EmberComponent`
+- `GlimmerComponent`
+- `Helper`
+- `Route`
+- `Controller`
+- `Service`
+- `Location`
+- `Resource`
+
+In general, the policy from this RFC forward will be that framework provided
+classes will use this method by default.
+
+### The `{{use}}` Helper
+
+Resources should also be usable without a component class, in cases where they
+don't exist, and for more dynamic use cases. The `{{use}}` helper allows
+resources to be setup based solely on the template.
+
+```hbs
+{{use SyncDocumentTitle @title}}
```

Comment:
So that example would actually be more like:

```hbs
{{#let (use this.counter) as |count|}}
  Count: {{count}}
{{/let}}
```

Since the `use` will return the `state` property of the resource directly, like the decorated property. Simple cases would probably work pretty well, but more complex cases would definitely get verbose. I think the "`let` for the rest of the template" idea could help here, potentially.